src/components/vote/ExecuteModal.tsx

```diff
@@ -0,0 +1,153 @@
+  const [attempting, setAttempting] = useState<boolean>(false)
+
+  // get theme for colors
+  const theme = useContext(ThemeContext)
+
+  // wrapper to reset state on modal close
+  function wrappedOndismiss() {
+    setHash(undefined)
+    setAttempting(false)
+    onDismiss()
+  }
+
+  async function onExecute() {
+    setAttempting(true)
+
+    // if callback not returned properly ignore
+    if (!executeCallback) return
+
+    // try delegation and store hash
+    const hash = await executeCallback(proposalId)?.catch((error) => {
+      setAttempting(false)
+      console.log(error)
+    })
+
+    if (hash) {
+      setHash(hash)
+    }
+  }
+
+  return (
+    <Modal isOpen={isOpen} onDismiss={wrappedOndismiss} maxHeight={90}>
+      {!attempting && !hash && (
+        <ContentWrapper gap="lg">
+          <AutoColumn gap="lg" justify="center">
+            <RowBetween>
+              <ThemedText.MediumHeader fontWeight={500}>
+                <Trans>Execute Proposal {proposalId}</Trans>
+              </ThemedText.MediumHeader>
+              <StyledClosed onClick={wrappedOndismiss} />
+            </RowBetween>
+            <RowBetween>
+              <ThemedText.Body>
+                <Trans>Executing this proposal will enact the calldata on-chain.</Trans>
+              </ThemedText.Body>
+            </RowBetween>
+            <ButtonPrimary onClick={onExecute}>
+              <ThemedText.MediumHeader color="white">
+                <Trans>Execute</Trans>
+              </ThemedText.MediumHeader>
+            </ButtonPrimary>
+          </AutoColumn>
+        </ContentWrapper>
+      )}
+      {attempting && !hash && (
+        <ConfirmOrLoadingWrapper>
+          <RowBetween>
+            <div />
+            <StyledClosed onClick={wrappedOndismiss} />
+          </RowBetween>
+          <ConfirmedIcon>
+            <CustomLightSpinner src={Circle} alt="loader" size={'90px'} />
+          </ConfirmedIcon>
+          <AutoColumn gap="100px" justify={'center'}>
+            <AutoColumn gap="12px" justify={'center'}>
+              <ThemedText.LargeHeader>
+                <Trans>Executing</Trans>
+              </ThemedText.LargeHeader>
+            </AutoColumn>
+            <ThemedText.SubHeader>
+              <Trans>Confirm this transaction in your wallet</Trans>
+            </ThemedText.SubHeader>
+          </AutoColumn>
+        </ConfirmOrLoadingWrapper>
+      )}
+      {hash && (
+        <ConfirmOrLoadingWrapper>
+          <RowBetween>
+            <div />
+            <StyledClosed onClick={wrappedOndismiss} />
+          </RowBetween>
+          <ConfirmedIcon>
+            <ArrowUpCircle strokeWidth={0.5} size={90} color={theme.primary1} />
+          </ConfirmedIcon>
+          <AutoColumn gap="100px" justify={'center'}>
+            <AutoColumn gap="12px" justify={'center'}>
+              <ThemedText.LargeHeader>
+                <Trans>Transaction Submitted</Trans>
```

Comment:
Instead of `Transaction Submitted`, can this specifically say `Execution submitted`?